behaviors/magic-button.js

```diff
@@ -30,10 +43,46 @@ const _ = require('lodash'),
       }
 
       return path;
+    },
+
+    /**
+     * transform rich text into a hyphen-delineated slug
+     * @param {string} data
+     * @returns {string}
+     */
+    toSlug: function (data) {
+      // remove EVERYTHING from the slug, then run it through speakingurl
+      return speakingurl(toPlainText(stripUnicode(data)), {
+        custom: {
+          _: '-' // convert underscores to hyphens
+        }
+      });
     }
   };
 
 /**
+ * Removes all unicode from string
+ * @param {string} str
+ * @returns {string}
+ */
+function stripUnicode(str) {
+  return str.replace(/[^A-Za-z 0-9\.,\?!@#\$%\^&\*\(\)-_=\+;:<>\/\\\|\}\{\[\]~]*/g, '');
+}
+
+/**
+ * remove all html stuff from a string
+ * @param {string} str
+ * @returns {string}
+ */
+function toPlainText(str) {
+  // coerce all text into a string. Undefined stuff is just an empty string
+  if (!_.isString(str)) {
```

Comment:
Cool. I'll keep it in mind for consistency's sake. Feel like ternarys are fine as long as their either semi-complex AND only one level deep (like the second half would be below.) Or 2 levels deep if all moving parts are simple.